tsdb/engine/tsm1/tombstone.go

```diff
@@ -79,34 +87,56 @@ func (t *Tombstoner) Delete() error {
 	if err := os.RemoveAll(t.tombstonePath()); err != nil {
 		return err
 	}
+	t.statsLoaded = false
 	return nil
 }
 
 // HasTombstones return true if there are any tombstone entries recorded.
 func (t *Tombstoner) HasTombstones() bool {
-	stat, err := os.Stat(t.tombstonePath())
-	if err != nil {
-		return false
-	}
-
-	return stat.Size() > 0
+	t.mu.RLock()
+	b := len(t.fileStats) > 0 && t.fileStats[0].Size > 0
```

Comment:
This should probably validate that `os.Stat` was at least called once, possibly checking `t.statsLoaded` first, then falling back to calling `TombstoneFiles` (or ensuring the cache gets populated)